File: calculator.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;
// Starter code
void calculator_with_undo() {
 int result = 0;
 vector<int> history_stack; // Store previous results
 
 while (true) {
   cout << "Current result: " << result << endl;
   cout << "Enter 'add [number]', 'subtract [number]', 'undo', or 'quit': ";
   
   string action;
   cin >> action;
   
   if (action == "add") {
     int number;
     cin >> number;
     //Task: Save current result, then add number
     history_stack.push_back(result); // Save current state
     result += number;
     
   } else if (action == "subtract") {
     int number;
     cin >> number;
     // Task: Save current result, then subtract number
     // Fill this in
     
   } else if (action == "undo") {
     // Task: Restore previous result
     // Fill this in
   } else if (action == "quit") {
       break;
   }
   }
  }
  int main() {
   calculator_with_undo();
   return 0;
  }

File: calculator.py
# Starter code
def calculator_with_undo():
 result = 0
 history_stack = [] # Store previous results
 
 while True:
   print(f"Current result: {result}")
   action = input("Enter 'add [number]', 'subtract [number]', 'undo', or 'quit': ").split()
   
   if action[0] == "add":
     # Task: Save current result, then add number
     history_stack.append(result) # Save current state
     result += int(action[1])
   
   elif action[0] == "subtract":
     # Task: Save current result, then subtract number
     pass # Fill this in
   
   elif action[0] == "undo":
     # Task: Restore previous result
     pass # Fill this in
     
   elif action[0] == "quit":
     break
